### PR TITLE
Upgrade Sonata to version 4

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

Part of #50 .

### CODE REVIEW

File: backend/.env [hunk 1]
```diff
@@ -16,7 +16,7 @@ DATABASE_URL=mysql://symfony:symfony@mysql:3306/symfony?serverVersion=5.7
 MAILER_DSN=smtp://mailhog:1025
 MAILER_FROM=noreply@localhost
 
-MESSENGER_TRANSPORT_DSN=sync://
+MESSENGER_TRANSPORT_DSN=doctrine://default
```
Comment (jordisala1991) [inline backend/.env:19]:
```suggestion
MESSENGER_TRANSPORT_DSN=sync://
```

File: backend/src/Admin/UserAdmin.php [hunk 2]
```diff
@@ -63,7 +63,7 @@ protected function configureFormOptions(array &$formOptions): void
     {
         $subject = $this->getSubject();
```
[the remark this answers is not in the document]
Comment (aerrasti) [reply backend/src/Admin/UserAdmin.php:64]: It does! Thanks i will check `hasSubject()` first :)

File: backend/src/Admin/UserAdmin.php [hunk 3]
```diff
@@ -102,6 +106,10 @@ protected function configureListFields(ListMapper $list): void
 
     protected function configureFormFields(FormMapper $form): void
     {
+        if (!$this->hasSubject()) {
```
Comment (jordisala1991) [inline backend/src/Admin/UserAdmin.php:109]: This particular case to me it is not correct this way. I would rather do:

```
$required = !$this->hasSubject() || null === $this->getSubject()->getId();
```

**Review by jordisala1991 — COMMENTED** [review]
Only a minor change, the rest looks great

**Review by aerrasti — APPROVED** [review]

**Pull request merged**